File: app/src/main/java/com/chooloo/www/callmanager/ui/cursor/CursorMvpPresenter.java
package com.chooloo.www.callmanager.ui.cursor;

import android.database.Cursor;
import android.os.Bundle;

import androidx.loader.content.Loader;

import com.chooloo.www.callmanager.ui.base.MvpPresenter;

public interface CursorMvpPresenter<V extends CursorMvpView> extends MvpPresenter<V> {
    void onRequestPermissionsResult(String[] permissions);

    void onRefresh();

    void onNoPermissions();

    void onResults();

    void onNoResults();

    Loader<Cursor> onCreateLoader(int id, Bundle args);

    void onLoadFinished(Loader<Cursor> loader, Cursor data);

    void onLoaderReset(Loader<Cursor> loader);

    void onEnablePermissionClick();
}

File: app/src/main/java/com/chooloo/www/callmanager/ui/cursor/CursorMvpPresenter.kt
package com.chooloo.www.callmanager.ui.cursor

import com.chooloo.www.callmanager.ui.base.MvpPresenter

interface CursorMvpPresenter<V : CursorMvpView?> : MvpPresenter<V> {
    fun onNoPermissions()
    fun onResults()
    fun onNoResults()
    fun onEnablePermissionClick()
}